docs/source/format/Flight.rst

```diff
@@ -153,6 +153,70 @@ A client that wishes to download the data would:
 
 .. _google.protobuf.Timestamp: https://protobuf.dev/reference/protobuf/google.protobuf/#timestamp
 
+Downloading Data by Running a Heavy Query
+-----------------------------------------
+
+A client may need to request a heavy query to download
+data. ``GetFlightInfo`` isn't returned until the requested query is
+completed. It means that the client is blocked. The client can use
+``PollFlightInfo`` instead of ``GetFlightInfo``:
```

Comment:
```suggestion
A client may need to request a heavy query to download
data. However, ``GetFlightInfo`` doesn't return until the query
completes, so the client is blocked. In this situation, the client
can use ``PollFlightInfo`` instead of ``GetFlightInfo``:
```